java/ql/lib/semmle/code/java/ControlFlowGraph.qll

```diff
@@ -1196,39 +1299,67 @@ private module ControlFlowGraphImpl {
       last(cc.getVariable(), n, completion) and result = first(cc.getBlock())
     )
     or
-    // Switch statements
-    exists(SwitchStmt switch | completion = NormalCompletion() |
-      // From the entry point control is transferred first to the expression...
-      n = switch and result = first(switch.getExpr())
-      or
-      // ...and then to one of the cases.
-      last(switch.getExpr(), n, completion) and result = first(switch.getACase())
+    // Switch statements and expressions
+    exists(StmtParent switch | switch instanceof SwitchStmt or switch instanceof SwitchExpr |
```

Comment:
It's a bit confusing to see this here when it looks like it's implied by all the disjuncts. Maybe lets move it into `getSwitchStatement`, so that predicate actually becomes restricted to switches?

Reply:
Have introduced `SwitchBlock` to do this in many places -- will follow up once this is merged to deduplicate predicates from SwitchStmt and SwitchExpr and use this more widely.